### PR TITLE
Fix gene relay query vars

### LINKED ISSUE #578: [Gene] Refine by medium not working
To repro:

1. Open any gene
2. Tap Refine
3. Select a medium

Works of all mediums are still shown; there doesn't seem to be any filtering.

### PR DESCRIPTION
**Pull request by orta**
Will comment inline

Fixes #578

### CODE REVIEW

File: src/lib/Components/ArtworkGrids/RelayConnections/GeneArtworksGrid.tsx [hunk 1]
```diff
@@ -63,6 +63,9 @@ const GeneArtworksGrid = createPaginationContainer(
         __id: props.gene.__id,
         count,
         cursor,
+        sort: props.sort,
+        medium: props.medium,
+        price_range: props.priceRange,
```
Comment (orta) [inline src/lib/Components/ArtworkGrids/RelayConnections/GeneArtworksGrid.tsx]: Tells the paginator to take these query vars into account

File: src/lib/Components/ArtworkGrids/RelayConnections/GeneArtworksGrid.tsx [hunk 2]
```diff
@@ -16,15 +16,15 @@ const GeneArtworksGrid = createPaginationContainer(
           cursor: { type: "String" }
           sort: { type: "String", defaultValue: "-partner_updated_at" }
           medium: { type: "String", defaultValue: "*" }
-          priceRange: { type: "String", defaultValue: "*-*" }
+          price_range: { type: "String", defaultValue: "*-*" }
```
Comment (orta) [inline src/lib/Components/ArtworkGrids/RelayConnections/GeneArtworksGrid.tsx]: I opted to move _all_ query vars to just use `price_range` as this is what metaphysics uses at the end (and what Eigen sends into Emission, and what force uses for hrefs)

File: src/lib/Containers/Gene.tsx [hunk 3]
```diff
@@ -193,15 +193,11 @@ export class Gene extends React.Component<Props, State> {
...
-            sort: newSettings.sort,
-          },
-          // TODO: is this param really required?
-          null
-        )
+        this.props.relay.refetch({
+          medium: newSettings.medium,
+          price_range: newSettings.selectedPrice,
+          sort: newSettings.sort,
+        })
```
Comment (orta) [inline src/lib/Containers/Gene.tsx]: made no difference

File: src/lib/Containers/Gene.tsx [hunk 4]
```diff
@@ -351,7 +347,8 @@ export default createRefetchContainer(
       ) {
       ...Header_gene
       ...About_gene
-      ...GeneArtworksGrid_gene
+      ...GeneArtworksGrid_gene @arguments(sort: $sort, medium: $medium, price_range: $price_range)
```
Comment (orta) [inline src/lib/Containers/Gene.tsx]: This container didn't pass down the options

File: src/lib/relay/QueryRenderers.tsx [hunk 5]
```diff
@@ -108,10 +108,10 @@ export const GeneRenderer: React.SFC<GeneRendererProps> = ({ render, geneID, med
   return (
     <QueryRenderer
       environment={environment}
-      query={graphql`
-        query QueryRenderersGeneQuery($geneID: String!) {
+      query={graphql.experimental`
+        query QueryRenderersGeneQuery($geneID: String!, $medium: String, $price_range: String) {
           gene(id: $geneID) {
-            ...Gene_gene
+            ...Gene_gene @arguments(medium: $medium, price_range: $price_range)
```
Comment (orta) [inline src/lib/relay/QueryRenderers.tsx]: This root container did not take the params and pass them in either

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  
<b>🃏 FAIL</b> in <a href="https://github.com/artsy/emission/blob/infinite_scroll/src/lib/Containers/__tests__/Gene-tests.tsx">Gene-tests.tsx</a>


<li>
<a href='https://github.com/artsy/emission/blob/infinite_scroll/src/lib/Containers/__tests__/Gene-tests.tsx#L88'>updates the state with new data from Eigen</a>
<br/>
Expected mock function to have been last called with: null as argument 2, but it was called with undefined.

<details>
<summary>Full message</summary>
<pre><code>
Error: expect(jest.fn()).lastCalledWith(expected)

Expected mock function to have been last called with:
  null as argument 2, but it was called with undefined.
    at /home/travis/build/artsy/emission/src/lib/Containers/__tests__/Gene-tests.tsx:88:25
    at tryCallOne (/home/travis/build/artsy/emission/node_modules/promise/lib/core.js:37:12)
    at /home/travis/build/artsy/emission/node_modules/promise/lib/core.js:123:15
    at flush (/home/travis/build/artsy/emission/node_modules/asap/raw.js:50:29)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
</code></pre>
</details>
</li>
<br/>


  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Containers/Gene.tsx (hunk 4 above)
Comment (alloy) [reply src/lib/Containers/Gene.tsx]: Ahhhh, yes

**Comment by alloy** [comment]
Looking good 👍

**Pull request merged**